packages/lab/src/layout/LayerLayout/LayerLayout.tsx

```diff
@@ -0,0 +1,111 @@
+import { forwardRef, HTMLAttributes, useState, useEffect } from "react";
+import cx from "classnames";
+
+import { makePrefixer, Breakpoints } from "@jpmorganchase/uitk-core";
+import { Scrim, ScrimProps } from "@jpmorganchase/uitk-lab";
+import "./LayerLayout.css";
+import { useChangeView } from "../utils";
+
+export const LAYER_POSITION = [
+  "center",
+  "left",
+  "top",
+  "right",
+  "bottom",
+] as const;
+
+export type LayerPosition = typeof LAYER_POSITION[number];
+
+export interface LayerLayoutProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Display or hide the scrim.
+   */
+  displayScrim?: boolean;
+  /**
+   * Defines the layer position within the screen.
+   */
+  position?: LayerPosition;
+  /**
+   * Breakpoint at which the layer will become fullscreen.
+   */
+  fullScreenAtBreakpoint?: keyof Breakpoints;
+  /**
+   * Disable all animations.
+   */
+  disableAnimations?: boolean;
+  /**
+   * Display or hide the component.
+   */
+  open?: boolean;
+  /**
+   * Props to be passed to the Scrim component.
+   */
+  scrimProps?: ScrimProps;
+}
+
+const withBaseName = makePrefixer("uitkLayerLayout");
+
+export const LayerLayout = forwardRef<HTMLDivElement, LayerLayoutProps>(
+  function LayerLayout(props, ref) {
+    const {
+      children,
+      className,
+      displayScrim = true,
+      position = "center",
+      fullScreenAtBreakpoint = "sm",
+      disableAnimations = false,
+      scrimProps,
+      open = true,
+      ...rest
+    } = props;
+
+    const [showComponent, setShowComponent] = useState(false);
+
+    useEffect(() => {
+      if (!open && disableAnimations) {
+        setShowComponent(false);
+      }
+
+      if (open && !showComponent) {
+        setShowComponent(true);
+      }
+    }, [open, showComponent]);
+
+    const fullScreen = useChangeView(fullScreenAtBreakpoint);
```

Comment:
Also noticing that when the `fullscreen` value changes the enter animation plays again

Reply:
Is this still an issue? I can't recreate it but I've also made some changes in the meantime so I'm not sure

Reply:
As discussed I'll have a catch up with Ben about the desired behaviour for this